### PR TITLE
MEN-3601: Add OpenSSL dependency on Debian package

### PR DESCRIPTION
**Pull request by lluiscampos**
Updated the build script and requirements, as well as the debian control
file shipped in the package.
    
Recipe for master differs now latest releases, so branch out these into
their own debian folder.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,6 +38,8 @@ RUN git checkout $MENDER_VERSION
 # Copy the debian recipe(s)
 COPY debian-master debian-master
 COPY debian-2.1.x debian-2.1.x
+COPY debian-2.2.x debian-2.3.x
+COPY debian-2.3.x debian-2.2.x
```
Comment (kacf) [inline Dockerfile:42]: The destination folder is opposite of the source folder. Typo?

Comment (lluiscampos) [reply Dockerfile:42]: Yes, typo :) Thanks!

Ironically one is a symlink to the other one, so technically doesn't matter. I'll fix.

**Comment by lluiscampos** [comment]
The acceptance tests require fixes (both here and in `mender-test-containers` repo). I am working on that.

**Comment by lluiscampos** [comment]
> The acceptance tests require fixes (both here and in `mender-test-containers` repo). I am working on that.

Fixed now; @kacf please have a look.

**Comment by lluiscampos** [comment]
@kacf I will launch an independent PR on `mender-test-containers` to fix the other repos, but for this one there was no fix required as the deb installation is done in the test code.

**Pull request merged**